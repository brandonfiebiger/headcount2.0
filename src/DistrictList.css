div {
  display: flex;
  flex-flow: row-wrap;
  flex-direction: column;
  width: 100vw;
}