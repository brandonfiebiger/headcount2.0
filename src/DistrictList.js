import React from 'react';
import DistrictCard from './DistrictCard';
import './DistrictList.css';

const DistrictList = ({ districts, searchedDistrict }) => {
  
  const districtCards = Object.keys(districts).map(district => {
    return <DistrictCard location={districts[district].location} stats={districts[district].stats} /> 
  })

  const searchedCards = searchedDistrict.map(district => <DistrictCard location={districts[district].location} stats={districts[district].stats} />)

  if (searchedDistrict.length) {
    return(
      <div>
        {searchedCards}
      </div>
    )
  }

  return(
    <div>
     {districtCards}
    </div>
  )
}



export default DistrictList;