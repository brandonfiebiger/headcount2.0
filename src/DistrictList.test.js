import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import DistrictList from './DistrictList';

describe('DistrictList', () => {

  let wrapper;
  let mockFn = jest.fn();
  let mockArray = [ 'COLORADO', 'COLORADO SPRINGS 11' ];

  beforeEach(() => {
    wrapper = shallow(<DistrictList addToCompare={mockFn} cardsToCompare={[]} districts={{}} searchedDistrict={[]} comparedObject={{}}/>);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should match the snapshot when there is length in the searchedDistrict array', () => {
    wrapper = shallow(<DistrictList addToCompare={mockFn} cardsToCompare={[]} districts={{}} searchedDistrict={mockArray} comparedObject={{}}/>)
    expect(wrapper).toMatchSnapshot();
  })
})