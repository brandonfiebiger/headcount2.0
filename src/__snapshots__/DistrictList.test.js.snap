// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DistrictList should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DistrictList
    addToCompare={[Function]}
    cardsToCompare={Array []}
    comparedObject={Object {}}
    districts={Object {}}
    searchedDistrict={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [],
        <div>
          
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [],
          <div>
            
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
